addons/udes_stock_refactoring/models/stock_move.py

```diff
@@ -359,6 +359,132 @@ def _refactor_action_by_maximum_quantity(self, maximum_qty):
+            max_range = len(grouped_moves)
+            # Build iterable of grouped moves to refactor, which can be passed into refactor_by_move_groups.
+            # Give each set of moves a unique group name so the pickings are kept separate.
+            for idx, moves_to_keep in enumerate(grouped_moves[:max_range]):
+                group_name = f"{picking.name}-{idx:0>3}"
+                moves_to_refactor.insert(0, (group_name, moves_to_keep))
+                all_moves |= moves_to_keep
+
+            all_moves |= moves.refactor_by_move_groups(moves_to_refactor)
+        return all_moves
+
+    def _split_and_group_by_weight(self, maximum_weight):
+        """
+        Split moves into groups of up to a maximum weight
+        :param maximum_weight: float weight to split and group moves
+        :returns: list of grouped moves
+        """
+        grouped_moves = []
+
+        remaining_moves = self
+
+        # See if any mls are equal to the maximum and add them as individual groups
+        exact_moves = self.filtered(
+            lambda l: (l.product_id.weight * l.product_uom_qty) == maximum_weight
+        )
+        remaining_moves -= exact_moves
+        for move in exact_moves:
+            grouped_moves.append(move)
+
+        # Split and group remaining moves by using moves_for_weight method, where if
+        # a move is split we add the split move to be split again and remove the move
+        # used from moves to be split. Add the moves to the grouped moves.
+        while remaining_moves:
+            moves_used, new_move, _weight = remaining_moves._moves_for_weight(maximum_weight)
+            if new_move:
+                remaining_moves |= new_move
+            remaining_moves -= moves_used
+            grouped_moves.append(moves_used)
+        return grouped_moves
+
+    def _moves_for_weight(self, weight, sort=True):
+        """
+        Return a subset of moves from self where their sum of the weight
+        to do is equal to parameter weight.
+        In case that a move needs to be split, the new move is
+        also returned (this happens when total weight * quantity in the move is
+        greater than weight parameter).
+        If there is not enough to do in the moves,
+        also return the remaining weight.
+        """
+        StockMove = self.env["stock.move"]
+        new_move = None
+        result = StockMove.browse()
+        if weight == 0:
+            return result, new_move, weight
+
+        if sort:
+            sorted_moves = self.sorted(
+                lambda m: m.product_id.weight * m.product_uom_qty, reverse=True
+            )
+            greater_equal_moves = sorted_moves.filtered(
+                lambda m: float_compare(
+                    (m.product_id.weight * m.product_uom_qty),
+                    weight,
+                    precision_rounding=m.product_uom.rounding,
+                )
+                >= 0
+            )
+            # last one will be at least equal
```

Comment:
Dont understand this comment, why should be last one at least equal ?

Reply:
I will re-word this

It was a copypasta from refactor by quantity.

All moves in `greater_equal_moves` will be at least equal (due to the previous filter) and the comment doesn't really give us any additional insight..

49f1f92fac55119ca6a74c7b18ee6ce583e988ce